src/foam/nanos/auth/AgentUserAuthService.java

```diff
@@ -100,6 +82,45 @@ public User actAs(X x, User entity) throws AuthenticationException {
     return user;
   }
 
+  public boolean canActAs(X x, User agent, User entity) {
+    try {
+      Group group = (Group) groupDAO_.find(entity.getGroup());
+      
+      if ( group == null ) {
+        throw new AuthorizationException("Entity must exist within a group.");
+      }
+      
+      if ( ! group.getEnabled() ) {
+        throw new AuthorizationException("Entity' group must be enabled.");
+      }
+      
+      /*
+      Finds the UserUserJunction object to see if user can act as the passed in user.
+      Source (agent) users are permitted to act as target (entity) users, not vice versa.
+      */
+      UserUserJunction permissionJunction = (UserUserJunction) agentJunctionDAO_.find(AND(
+        EQ(UserUserJunction.SOURCE_ID, agent.getId()),
+        EQ(UserUserJunction.TARGET_ID, entity.getId())
+        ));
```

Comment:
The indentation is one level too deep starting on this line and ending on line 116.